### ISSUE TITLE
atomic fails silently on `AssertionError`

### ISSUE THREAD

**Issue by mtrmac** [projectatomic/atomic#928]
To reproduce, right now:
```sh
ATOMIC_CONF=tests/unit/fixtures/atomic.conf ATOMIC_CONFD=/var/empty/ sudo -E ./atomic sign --sign-by=mitr@volny.cz -d . docker.io/library/busybox:latest
```

This causes an `AssertionError` in `RegistryInspect.get_manifest`; per language spec, equivalent to `raise AssertionError`, i.e. with empty `args`.

This is ultimately caught in the top-level
```python
    except Exception as e: # pylint: disable=broad-except
        write_err("%s" % str(e))
        sys.exit(1)
```

but because `e.args` is `()`, `str(e)` is empty string. The net result is that this writes an empty line to `stderr`, and exits with a non-zero exit code, with no other indication what has happened.

Adding `--debug` does print the full traceback, due to the top-level `finally:` block:
```
Traceback (most recent call last):
  File "./atomic", line 189, in <module>
    sys.exit(_func())
  File "/home/mitr/t/atomic/Atomic/sign.py", line 79, in sign
    manifest = ri.get_manifest()
  File "/home/mitr/t/atomic/Atomic/discovery.py", line 41, in get_manifest
    assert(self.fqdn is not None)
AssertionError
```

I’m not entirely sure what the cleanest fix is.

Perhaps add an `except AssertionError` clause always printing a traceback (because that’s the only information available), and then somehow surpress the traceback print in `finally:`.

Or perhaps make the `except Exception` more general, and print a traceback in there if `str(e)` is empty, to ensure that _something_ is always printed.

Or perhaps something else entirely.

**Comment by baude** [comment]
@mtrmac i think is fixed now yes?  can we close?

**Comment by mtrmac** [comment]
The cause of _that particular_ `AssertionError` is fixed in #932.

The _general_ handling of `AssertionError` is still to fail without any message.  To test, add e.g. `assert(False)` to `RegistryInspect.get_manifest` and cause it to be called.  As described above, the result is an empty line on output, without any text, and `--debug` is necessary to see a traceback.

**Comment by baude** [comment]
Fixed in https://github.com/projectatomic/atomic/pull/959
